src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -17,10 +17,7 @@ OutputStateMachineEngine::OutputStateMachineEngine(std::unique_ptr<ITermDispatch
     _pTtyConnection(nullptr),
     _lastPrintedChar(AsciiChars::NUL)
 {
-}
-
-OutputStateMachineEngine::~OutputStateMachineEngine()
-{
+    THROW_IF_NULL_ALLOC(_dispatch.get());

```

Comment:
This isn't exactly correct, it'll result in a hresult exception `E_OUTOFMEMORY` or whatever. It's really more of an invalid parameter, but also it's more of a "we could avoid the throwing constructor completely" case?

Reply:
Is there a `gsl::not_null_but_also_a_unique_ptr`?

Reply:
well it should be THROW_HR_IF(E_INVALIDARG, _dispatch.get()) probably.

Reply:
But that's all over the codebase. So I should bulk do that.

Reply:
This is now #4099